src/main/java/org/opensearch/ad/rest/RestDeleteAnomalyDetectorAction.java

```diff
@@ -16,67 +16,130 @@
+import java.util.function.Function;
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.delete.DeleteResponse;
+import org.opensearch.ad.AnomalyDetectorExtension;
 import org.opensearch.ad.AnomalyDetectorPlugin;
 import org.opensearch.ad.constant.CommonErrorMessages;
 import org.opensearch.ad.rest.handler.AnomalyDetectorActionHandler;
+import org.opensearch.ad.settings.AnomalyDetectorSettings;
 import org.opensearch.ad.settings.EnabledSetting;
 import org.opensearch.ad.transport.DeleteAnomalyDetectorAction;
 import org.opensearch.ad.transport.DeleteAnomalyDetectorRequest;
-import org.opensearch.client.node.NodeClient;
-import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.json.JsonXContent;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.extensions.rest.ExtensionRestResponse;
 import org.opensearch.rest.RestRequest;
-import org.opensearch.rest.action.RestToXContentListener;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.sdk.BaseExtensionRestHandler;
+import org.opensearch.sdk.ExtensionsRunner;
+import org.opensearch.sdk.RouteHandler;
+import org.opensearch.sdk.SDKClient.SDKRestClient;
+import org.opensearch.sdk.SDKClusterService;
+import org.opensearch.transport.TransportService;
 
 import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
 
 /**
  * This class consists of the REST handler to delete anomaly detector.
  */
-public class RestDeleteAnomalyDetectorAction extends BaseRestHandler {
+public class RestDeleteAnomalyDetectorAction extends BaseExtensionRestHandler {
 
     public static final String DELETE_ANOMALY_DETECTOR_ACTION = "delete_anomaly_detector";
 
     private static final Logger logger = LogManager.getLogger(RestDeleteAnomalyDetectorAction.class);
     private final AnomalyDetectorActionHandler handler = new AnomalyDetectorActionHandler();
 
-    public RestDeleteAnomalyDetectorAction() {}
+    @Inject
+    private NamedXContentRegistry namedXContentRegistry;
+    private Settings settings;
+    private TransportService transportService;
+    @Inject
+    private SDKRestClient client;
+    @Inject
+    private SDKClusterService clusterService;
+    private ExtensionsRunner extensionsRunner;
+
+    public RestDeleteAnomalyDetectorAction(ExtensionsRunner extensionsRunner, SDKRestClient client) {
+        this.extensionsRunner = extensionsRunner;
+        this.namedXContentRegistry = extensionsRunner.getNamedXContentRegistry().getRegistry();
+        this.settings = extensionsRunner.getEnvironmentSettings();
+        this.transportService = extensionsRunner.getExtensionTransportService();
+        this.client = client;
+
+        this.clusterService = new SDKClusterService(extensionsRunner);
```

Comment:
I see that we're injecting the clusterService and creating a new cluster service object here but it's not used anywhere